draft-regext-rdap-extensions.md

```diff
@@ -685,8 +694,63 @@ consultation of the definition of "fizzbuzz1" will determine its
 relationship with "fizzbuzz0". Additionally, "fizzbuzz99" may be the
 predecessor of "fizzbuzz0".
 
-An RDAP extension definition MUST explicitly denote its compliance with any
-versioning scheme, such as [@?I-D.ietf-regext-rdap-versioning].
+If an RDAP extension uses a versioning method, such as [@?I-D.ietf-regext-rdap-versioning],
+it MUST be explicitly described in its specification.
+
+### Breaking Changes in Successors {#breaking_changes}
+
+With the current extension model, an extension with a
+successor with breaking changes is indistinguishable from a new,
+unrelated extension.  Additionally, there is no signaling
+mechanism in RDAP to specify successors with breaking changes.
+Implementers of such changes should consider the following:
+
+ - whether the new version of the extension can be provided alongside
+   the old version of the extension, so that a service can simply
+   support both during a transition period;
+ - whether some sort of client signaling should be supported, so that
+   clients can opt for the old or new version of the extension in
+   responses that they receive (see
+   [@?I-D.ietf-regext-rdap-x-media-type] for an example of how this
+   might work); and
+ - whether the extension itself should define how versioning is
+   handled within the extension documentation.
+
+When using a transition period between two versions of an extension by
+using both versions, the successor must not conflict with the predecessor.
+Typically, this is not an issue when the rules of RDAP namespaced identifiers
+are followed (see #(bare_extensions)), but care should be taken if the
+extensions specify other behaviors not protected by namespaces, particularly
+referrals (see (#referrals)).
+
+Another breaking change is to introduce a new object class where a client
+previously expected another, such as:
+
+ - a query using a path and/or query parameters
+   (e.g., /domain/foo.example produces "new_domain" instead of "domain");
+ - a referral (e.g., "related" produces "new_domain" instead of "domain");
+ - search results; and
+ - other JSON arrays and JSON members.
+
+Breaking changes may occur in requirements for processing of data in
+protocol elements that appear in both a successor and predecessor.
+For example, a profile extension (see (#profiles)) may require domain names
+always end with a dot ("."). Should its successor remove this requirement
+this could be considered a breaking change.
+
+### Non-breaking Changes in Successors {#nonbreaking_changes}
+
+The following are considered non-breaking changes between a successor and
+a predecessor.
+
+ - new referrals
+ - new JSON members
+ - new query paths
+ - new query parameters
+
+The use of a new HTTP header (i.e., one previously not in-use with the predecessor),
+may either be a breaking change or non-breaking change, depending on the usage of
```

Comment:
"may either be a breaking change or non-breaking change" -> "may either be a breaking change or a non-breaking change"